# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1366182570 14400
#      Wed Apr 17 03:09:30 2013 -0400
# Node ID 67d4111d078db76ffc7b92758971e7e3add8fc93
# Parent  f3a63fdbd725244c2861d226d8a5b9733e1fcb8c
look for save/load files starting in current directory

* main-window.cc (main_window::handle_save_workspace_request,
main_window::handle_load_workspace_request): Start file dialog widget
in current directory instead of home directory.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -126,32 +126,32 @@ void
 main_window::report_status_message (const QString& statusMessage)
 {
   status_bar->showMessage (statusMessage, 1000);
 }
 
 void
 main_window::handle_save_workspace_request (void)
 {
-  QString selectedFile =
-    QFileDialog::getSaveFileName (this, tr ("Save Workspace As"),
-                                  resource_manager::get_home_path ());
-  if (!selectedFile.isEmpty ())
+  QString file =
+    QFileDialog::getSaveFileName (this, tr ("Save Workspace As"), ".");
+
+  if (! file.isEmpty ())
     octave_link::post_event (this, &main_window::save_workspace_callback,
-                             selectedFile.toStdString ());
+                             file.toStdString ());
 }
 
 void
 main_window::handle_load_workspace_request (const QString& file_arg)
 {
   QString file = file_arg;
 
   if (file.isEmpty ())
-    file = QFileDialog::getOpenFileName (this, tr ("Load Workspace"),
-                                         resource_manager::get_home_path ());
+    file = QFileDialog::getOpenFileName (this, tr ("Load Workspace"), ".");
+
   if (! file.isEmpty ())
     octave_link::post_event (this, &main_window::load_workspace_callback,
                              file.toStdString ());
 }
 
 void
 main_window::handle_clear_workspace_request (void)
 {
